### PR TITLE
Add google analytics key to tardis docs

### PR DESCRIPTION
**Pull request by KasukabeDefenceForce**
### :pencil: Description

**Type:** :beetle: `bugfix` | :rocket: `feature` | :biohazard: `breaking change` | :vertical_traffic_light: `testing` | :memo: `documentation` | :roller_coaster: `infrastructure`

Add google analytics to tardis docs

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Review by atharva-2001 — APPROVED** [review]

**Comment by atharva-2001** [comment]
The docs failed for some reason. We should already have sphinx, right? The google analytics is only thing that is not in the environment

**Comment by KasukabeDefenceForce** [comment]
> The docs failed for some reason. We should already have sphinx, right? The google analytics is only thing that is not in the environment

The docs are failing because it is using the conf.py from the PR to build docs but it is using the workflow file from the master.

**Comment by KasukabeDefenceForce** [comment]
> The docs failed for some reason. We should already have sphinx, right? The google analytics is only thing that is not in the environment

There is some issue with actions, because it is not running the steps to install the google analytics extension even though the workflow file has that step.

**Comment by atharva-2001** [comment]
That is because this is a `pull_request_target` event workflow. Edit the workflow triggers and make it work on your fork. Once it runs you can share a link here and undo the steps so that just the relevant ones remain.

**Comment by KasukabeDefenceForce** [comment]
> That is because this is a `pull_request_target` event workflow. Edit the workflow triggers and make it work on your fork. Once it runs you can share a link here and undo the steps so that just the relevant ones remain.

https://github.com/KasukabeDefenceForce/tardis/actions/runs/10004874599/job/27654456837?pr=31
Is this what you asked for?

**Review by atharva-2001 — APPROVED** [review]

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -103,6 +103,10 @@ jobs:
       - name: Install package
         run: pip install -e .
 
+      - name: Install Sphinx extensions
+        run: |
+          pip install sphinxcontrib-googleanalytics
```
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:108]: we should at some point add this to env file, but we will need to ask if other members are happy with this

**Review by jvshields — APPROVED** [review]

**Pull request merged**